src/universal/redux/storageMerger.js

```diff
@@ -9,26 +7,22 @@ export default function merger(initialState, persistedState) {
     const reducerName = persistedStateKeys[i];
     if (!whitelist.has(reducerName)) continue;
     const value = persistedState[reducerName];
-    if (reducerName === 'auth' && value) {
-      // Always perform a fresh decode for security's sake:
-      let authTokenObj = null;
-      try {
-        authTokenObj = jwtDecode(value.token);
-      } catch (e) {
-        console.warn(`unable to decode jwt: ${e}`);
-        continue;
-      }
-      if (authTokenObj.exp < Date.now() / 1000) {
+    if (reducerName === 'auth' && value && value.obj) {
+      if (value.obj.exp < Date.now() / 1000) {
         continue;
       }
     }
     if (reducerName === 'cashay') {
-      if (value && value.entities) {
-        // cashay now supports rehydrating sub data
+      if (value) {
+        const {User} = value.entities || {};
+        value.entities = User ? {User} : {};
+        value.result = {};
+
+        // cashay now supports rehydrating sub data, but this can't account for removed items
         // remove items with unbounded growth in production
-        value.result.presence = undefined;
-        value.entities.Presence = undefined;
-        value.entities.Task = undefined;
+        // value.result.presence = undefined;
```

Comment:
@mattkrick do you need a different comment here now? Were these supposed to be commented out?